File: backend/src/main/java/magazyn/dto/ZamowienieKlientaDTO.java
package magazyn.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ZamowienieKlientaDTO {
    private Integer id;
    private OffsetDateTime data;
    private String imieKlienta;
    private String nazwiskoKlienta;
    private Integer status;
    private List<PozycjaZamowieniaDTO> produkty;

    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    public static class PozycjaZamowieniaDTO {
        private Integer produktId;
        private String nazwaProduktu;
        private Integer ilosc;
        private String kodKreskowy;
    }
}

File: frontend/app/src/main/java/com/example/magazyn/api/dtos/ZamowienieKlientaDTO.kt
package com.example.magazyn.api.dtos

data class ZamowienieKlientaDTO(
    val id: Int,
    val data: String,
    val imieKlienta: String,
    val nazwiskoKlienta: String,
    val status: Int,
    val produkty: List<PozycjaZamowieniaDTO>
) {
    data class PozycjaZamowieniaDTO(
        val produktId: Int,
        val nazwaProduktu: String,
        val ilosc: Int,
        val kodKreskowy: String
    )
}
